services/web/server/src/simcore_service_webserver/products_events.py

```diff
@@ -32,6 +32,12 @@ async def setup_product_templates(app: web.Application):
         # cleanup
 
 
+def _set_app_state(app: web.Application, app_products: dict[str, Product]):
+    app[APP_PRODUCTS_KEY] = app_products
+    # default is the first of a sorted priority-based list
+    app[f"{APP_PRODUCTS_KEY}_default"] = next(iter(app_products.values())).name
```

Comment:
Why not use the same function as the catalog and the api-server?
It would be better to ably have one way of retrieving the the default product. In this case the service will fail differently from the others, if the product is not defined.

Reply:
yep. you are right.